akka-actor-tests/src/test/scala/akka/pattern/BackoffOnRestartSupervisorSpec.scala

```diff
@@ -246,5 +246,39 @@ class BackoffOnRestartSupervisorSpec extends AkkaSpec with ImplicitSender {
         probe.expectTerminated(supervisor)
       }
     }
+
+    "keep restarting when failures happen outside of withinTimeRange" in {
+      val probe = TestProbe()
+
+      val backoffProps = BackoffSupervisor.props(
+        BackoffOpts
+          .onFailure(
+            TestActor.props(probe.ref),
+            "child",
+            minBackoff = 100.millis,
+            maxBackoff = 800.millis,
+            randomFactor = 0.0)
+          // really short withinTimeRange should reset pretty much after every restart
+          .withSupervisorStrategy(OneForOneStrategy(3, withinTimeRange = 10.millis)(SupervisorStrategy.defaultDecider)))
+      val supervisor = system.actorOf(backoffProps, "supervisor")
+      probe.watch(supervisor)
+
+      // since the backoff is 100.millis we know the
+      // 10 millis pass between _every_ restart
+      probe.expectMsg("STARTED")
+      supervisor ! "THROW"
+      probe.expectMsg("STARTED")
+      supervisor ! "THROW"
+      probe.expectMsg("STARTED")
+      supervisor ! "THROW"
+      probe.expectMsg("STARTED")
+      supervisor ! "THROW"
```

Comment:
any logging that should be silenced here?

Reply:
Better up, added WithLogCapturing and kept logging at debug for useful info on failures.